server/dbData.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * @fileoverview database access object for data.
+ */
+
+var mongodb = require('mongodb');
+var MongoClient = mongodb.MongoClient;
+var mongoUrl = 'mongodb://localhost:27017/express';
+
+var log = require('./log');
+
+/** @type {dbData} */
+module.exports = dbData;
+
+/**
+ * @constructor
+ */
+function dbData() {}
+
+/**
+ * @private @const {string}
+ */
+dbData.DATA_COLLECTION_ = 'file';
+
+/**
+ * @private @const {string}
+ */
+dbData.PROGRESS_COLLECTION_ = 'uploadProgress';
+
+/**
+ * @private @type {string}
+ */
+dbData.USERNAME_ = 'anonymous';
+
+/**
+ * @typedef {{
+ *   error: string
+ * }}
+ */
+dbData.Error;
+
+/**
+ * Creates a connection to the mongodb.
+ * @param {function(!mongodb.Db, Object=)} callback The callback function.
+ */
+dbData.createConnection = function(callback) {
+  MongoClient.connect(mongoUrl, function(err, db) {
+    callback(db, err);
+  });
+};
+
+/**
+ * Inserts a new file into database.
+ * @param {!mongodb.Db} db The database.
+ * @param {string} path Path to the folder of the file.
+ * @param {string} fileName File name of the file.
+ * @param {!Object} property Properties of the file.
```

Comment:
Can you list the attributes of this Object? looks like you are inserting this to db so we must know what exactly the attributes are.